lib/connectors/base/connector.rb

```diff
@@ -37,7 +37,19 @@ def initialize(configuration: {})
         @configuration = configuration.dup || {}
       end
 
-      def yield_documents; end
+      def yield_documents(job_description = {}); end
```

Comment:
This was guidance from me. Our connectors instances are initialized _before_ we've claimed the `.elastic-connectors-sync-job` record. But it makes sense that a given call to `yield_documents` should be done within the context of a specific job. This will become even more true with the work that @vidok is proposing for the Job Processing changes.